packages/errors/src/test-utils/expect-to-throw-fuel-error.test.ts

```diff
@@ -2,67 +2,72 @@ import { FuelError } from '..';
 
-it('compares whole or partial error objects (sync)', () => {
-  expectToThrowFuelError(throwFn, error);
-  expectToThrowFuelError(throwFn, { code: error.code });
-});
+  const throwFn = () => {
+    throw error;
+  };
 
-it('compares whole or partial error objects (async)', async () => {
-  await expectToThrowFuelError(asyncThrowFn, error);
-  await expectToThrowFuelError(asyncThrowFn, { code: error.code });
-});
+  const asyncThrowFn = async () => new Promise(throwFn);
 
-it('fails when passed fn doesnt throw (sync)', () => {
-  expect(() => expectToThrowFuelError(() => {}, error)).toThrow(/Passed-in lambda didn't throw./);
-});
+  it('compares whole or partial error objects (sync)', () => {
+    expectToThrowFuelError(throwFn, error);
+    expectToThrowFuelError(throwFn, { code: error.code });
+  });
 
-it('fails when passed fn doesnt throw (async)', async () => {
-  await expect(expectToThrowFuelError(async () => {}, error)).rejects.toThrow(
-    /Passed-in lambda didn't throw./
-  );
-});
+  it('compares whole or partial error objects (async)', async () => {
+    await expectToThrowFuelError(asyncThrowFn, error);
+    await expectToThrowFuelError(asyncThrowFn, { code: error.code });
+  });
 
-it('fails when partial error object doesnt contain code (sync)', () => {
-  expect(() =>
-    expectToThrowFuelError(
-      throwFn,
-      // @ts-expect-error code property is required per type definition
-      { message: error.message }
-    )
-  ).toThrow();
-});
+  it('fails when passed fn doesnt throw (sync)', () => {
+    const fn = expectToThrowFuelError(() => {}, error);
+    expect(fn).rejects.toThrow(/Passed-in lambda didn't throw./);
+  });
 
-it('fails when partial error object doesnt contain code (async)', async () => {
-  await expect(
-    expectToThrowFuelError(
-      asyncThrowFn,
-      // @ts-expect-error code property is required per type definition
-      { message: error.message }
-    )
-  ).rejects.toThrow();
-});
+  it('fails when passed fn doesnt throw (async)', () => {
+    const fn = expectToThrowFuelError(async () => {}, error);
+    expect(fn).rejects.toThrow(/Passed-in lambda didn't throw./);
+  });
 
-it('fails with non-fuel errors (sync)', () => {
-  const expected = { code: FuelError.CODES.PARSE_FAILED };
-  const thrower = () => {
-    throw new Error('x'); // not a FuelError
-  };
+  it('fails when expected error object doesnt contain code (sync)', () => {
+    const expectedError = { message: error.message }; // no code here
+    const fn = () => expectToThrowFuelError(throwFn, expectedError as FuelError);
+    expect(fn).rejects.toThrow('Expected error must contain a code.');
+  });
 
-  expect(() => expectToThrowFuelError(thrower, expected)).toThrow();
-});
+  it('fails when partial error object doesnt contain code (async)', () => {
+    const expectedError = { message: error.message }; // no `code` here
+    const fn = () => expectToThrowFuelError(asyncThrowFn, expectedError as FuelError);
```

Comment:
The benefit of using 
```ts
expect(
    expectToThrowFuelError(
      asyncThrowFn,
      // @ts-expect-error code property is required per type definition
      { message: error.message }
    )
```
instead of 
```ts
const fn = () => expectToThrowFuelError(asyncThrowFn, expectedError as FuelError);
```
with regards to `expectedError`  that in the first case the `// ts-expect-error` is also testing the type system's requirements that a `code` property must be provided, whereas casting it directly to `FuelError` doesn't test that.g

Reply:
TBH, I find `@ts-*` tags an ugly thing. But you have a point.